lib/ejs.js

```diff
@@ -753,8 +762,20 @@ Template.prototype = {
             self.source += includeSrc;
             self.dependencies.push(exports.resolveInclude(include[1],
               includeOpts.filename));
+            includeObj.dependencies.forEach(function(dep) {
+              self.dependencies.push(dep);
+            });
             return;
           }
+        } else if (opts.legacyInclude && (include = line.match(/^\s*include\(\s*['"](\S+)['"].*\)/))) {
```

Comment:
The&nbsp;`include(…)`&nbsp;function is&nbsp;not&nbsp;legacy, and&nbsp;it’s&nbsp;meant to&nbsp;only&nbsp;be&nbsp;resolved at&nbsp;runtime, unlike&nbsp;the&nbsp;legacy&nbsp;`include`&nbsp;directive, which&nbsp;is&nbsp;inefficiently&nbsp;resolved at&nbsp;compile&nbsp;time.

Reply:
Ok, i reverted this change.
I'll squash the branch's commit later.

Reply:
Thanks for catching this. Would not want to remove all the functionality all together. :)